enterprise/api/v1/views.py

```diff
@@ -242,6 +253,128 @@ def get_serializer_class(self):
         return serializers.EnterpriseCourseEnrollmentWriteSerializer
 
 
+class LicensedEnterpriseCourseEnrollmentViewSet(EnterpriseReadOnlyModelViewSet):
+    """
+    API views for the ``licensed-enterprise-course-enrollment`` API endpoint.
+    """
+
+    queryset = models.LicensedEnterpriseCourseEnrollment.objects.all()
+    serializer_class = serializers.LicensedEnterpriseCourseEnrollmentReadOnlySerializer
+
+    def _validate_license_revoke_data(self, request_data):
+        """
+        Ensures the request data contains the necessary information.
+
+        Arguments:
+            request_data (dict): A dictionary of data passed to the request
+        """
+        user_id = request_data.get('user_id')
+        enterprise_id = request_data.get('enterprise_id')
+
+        if not user_id or not enterprise_id:
+            msg = 'user_id and enterprise_id must be provided.'
+            return Response(msg, status=status.HTTP_400_BAD_REQUEST)
+
+        return None
+
+    @action(methods=['post'], detail=False)
+    @permission_required('enterprise.can_access_admin_dashboard', fn=lambda request: request.data.get('enterprise_id'))
+    def license_revoke(self, request, *args, **kwargs):
+        """
+        Changes the mode for a user's licensed enterprise course enrollments to the "audit" course mode.
+        """
+        if get_course_overviews is None or get_certificate_for_user is None:
+            raise NotConnectedToOpenEdX(
+                _('To use this endpoint, this package must be '
+                  'installed in an Open edX environment.')
+            )
+
+        request_data = request.data.copy()
+        self._validate_license_revoke_data(request_data)
+
+        user_id = request_data.get('user_id')
+        enterprise_id = request_data.get('enterprise_id')
+        mode = CourseModes.AUDIT
+
+        enterprise_customer_user = get_object_or_404(
+            models.EnterpriseCustomerUser,
+            user_id=user_id,
+            enterprise_customer=enterprise_id,
+        )
+
+        licensed_enrollments = self.queryset.filter(
+            enterprise_course_enrollment__enterprise_customer_user=enterprise_customer_user
+        )
+        if not licensed_enrollments:
+            msg = 'User {} does not have any licensed enterprise course enrollments.'.format(user_id)
+            return Response(msg, status=status.HTTP_404_NOT_FOUND)
```

Comment:
Do we want to consider revoking an enterprise license from someone who hasn't enrolled in anything yet as an error? I think it's a valid scenario for someone to have a license revoked before they enrolled in anything. I wish HTTP had a "we successfully didn't do anything but that's what was supposed to happen in this case" type response status.

Reply:
Yeah, I guess there's no harm is letting the rest of the method execute in this case, since `licensed_enrollments` would be an empty list with nothing to iterate through anyways.

I agree a revoke is possible before they'd actually enrolled in anything. I will update.